drivers/adc/adc_mcux_adc16.c

```diff
@@ -19,19 +24,80 @@ LOG_MODULE_REGISTER(adc_mcux_adc16);
 
 struct mcux_adc16_config {
 	ADC_Type *base;
+#ifndef CONFIG_ADC_MCUX_ADC16_ENABLE_EDMA
 	void (*irq_config_func)(const struct device *dev);
+#endif
+	uint32_t clk_source;	/* ADC clock source selection */
+	uint32_t long_sample;	/* ADC long sample mode selection */
+	uint32_t hw_trigger_src;  /* ADC hardware trigger source */
+				/* defined in SIM module SOPT7 */
+	uint32_t dma_slot;	/* ADC DMA MUX slot */
```

Comment:
Nit, but is dma_slot need if CONFIG_ADC_MCUX_ADC16_ENABLE_EDMA=n?

Reply:
yes not need, I will remove it